helix-term/src/ui/prompt.rs

```diff
@@ -355,15 +360,36 @@ impl Prompt {
             }
         }
 
-        if let Some(doc) = (self.doc_fn)(&self.line) {
-            let mut text = ui::Text::new(doc.to_string());
+        if let Some(PromptDoc { doc, aliases }) = (self.doc_fn)(&self.line) {
+            let mut string = String::with_capacity(
+                doc.len() + aliases.iter().fold(0, |acc, alias| acc + alias.len() + 2) + 12,
+            );
+            string.push_str(doc);
+
+            if !aliases.is_empty() {
+                string.push('\n');
+                string.push_str("Aliases: ");
+                for alias in aliases {
+                    string.push_str(alias);
+                    string.push_str(", ");
+                }
+                string.pop();
+                string.pop();
+            }
+
+            let width = BASE_WIDTH * 3;
+            let height = 2 + string
+                .lines()
+                .fold(0, |acc, s| acc + (s.len() as u16 / width) + 1);
+
+            let mut text = ui::Text::new(string);
```

Comment:
How about using the paragraph widget instead? It can calculate the total height / width of the paragraph too: https://github.com/helix-editor/helix/blob/master/helix-tui/src/widgets/paragraph.rs